src/wasm/wasm-type.cpp

```diff
@@ -65,6 +71,12 @@ struct TypeInfo {
 
   bool isNullable() const { return kind == RefKind && ref.nullable; }
 
+  // If this TypeInfo represents a Type that can be represented more simply, set
+  // `out` to be that simpler Type and return true. For example, this handles
+  // canonicalizing the TypeInfo representing (ref null any) into the BasicType
+  // anyref. It also handles eliminating singleton tuple types.
+  bool getCanonical(Type& out) const;
```

Comment:
Could this perhaps return `none` instead of false? The out param + bool adds some complexity here. But I guess that's not possible if you want this to turn `none` into something or something into `none` - maybe not though?

Reply:
Ideally it return a `std::optional<Type>`, but until we have C++17 that seems better. I think we could use `Type::none` in principle, but the fact that I had to think through that for a bit make me think that it would be better not to do so.